dynawo/sources/Models/CPP/ModelNetwork/DYNModelCurrentLimits.cpp

```diff
@@ -83,36 +89,58 @@ ModelCurrentLimits::evalG(const double& t, const double& current, state_g* g, co
   }
 }
 
+constraints::ConstraintData
+ModelCurrentLimits::constraintData(const std::string& kind, unsigned int i) {
+  // The value for the limit and the current in SI units (Amperes)
+  bool isTemporary = openingAuthorized_[i];
+  if (isTemporary) {
+    return constraints::ConstraintData(kind, limits_[i]*factorPuToA_, lastCurrentValue_*factorPuToA_, side_, acceptableDurations_[i]);
+  } else {
+    return constraints::ConstraintData(kind, limits_[i]*factorPuToA_, lastCurrentValue_*factorPuToA_, side_);
+  }
+}
+
 ModelCurrentLimits::state_t
 ModelCurrentLimits::evalZ(const string& componentName, const double& t, state_g* g, ModelNetwork* network, const double& desactivate,
     const string& modelType) {
   state_t state = ModelCurrentLimits::COMPONENT_CLOSE;
+  using constraints::ConstraintData;
 
   for (unsigned int i = 0; i < limits_.size(); ++i) {
     if (!(desactivate > 0)) {
       if (g[0 + 2 * i] == ROOT_UP && !activated_[i]) {
         if (openingAuthorized_[i]) {  // Delay is specified => temporary limit
-          DYNAddConstraint(network, componentName,  true,  modelType, OverloadUp, acceptableDurations_[i], side_);
+          DYNAddConstraintWithData(network, componentName, true, modelType,
+            constraintData("OverloadUp", i),
+            OverloadUp, acceptableDurations_[i], side_);
         } else {
-          DYNAddConstraint(network, componentName,  true,  modelType, IMAP, side_);
+          DYNAddConstraintWithData(network, componentName, true, modelType,
+            constraintData("IMAP", i),
+            IMAP, side_);
         }
         tLimitReached_[i] = t;
         activated_[i] = true;
       }
 
       if (g[0 + 2 * i] == ROOT_DOWN && activated_[i]) {
         if (openingAuthorized_[i]) {  // Delay is specified => temporary limit
-          DYNAddConstraint(network, componentName,  false,  modelType, OverloadUp, acceptableDurations_[i], side_);
+          DYNAddConstraintWithData(network, componentName, false, modelType,
+            constraintData("OverloadUp", i),
+            OverloadUp, acceptableDurations_[i], side_);
         } else {
-          DYNAddConstraint(network, componentName,  false,  modelType, IMAP, side_);
+          DYNAddConstraintWithData(network, componentName, false, modelType,
+            constraintData("IMAP", i),
```

Comment:
please, change IMAP to PATL, everywhere else

Reply:
Renamed everywhere